infoschema/tiniub/tables.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tiniub
+
+import (
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/domain"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/meta/autoid"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/types"
+)
+
+type slowQueryTable struct {
+	meta *model.TableInfo
+	cols []*table.Column
+}
+
+func tableFromMeta(alloc autoid.Allocator, meta *model.TableInfo) (table.Table, error) {
+	return createSlowQueryTable(meta), nil
+}
+
+// createSlowQueryTable creates all slowQueryTables
+func createSlowQueryTable(meta *model.TableInfo) *slowQueryTable {
+	columns := make([]*table.Column, 0, len(meta.Columns))
+	for _, colInfo := range meta.Columns {
+		col := table.ToColumn(colInfo)
+		columns = append(columns, col)
+	}
+	t := &slowQueryTable{
+		meta: meta,
+		cols: columns,
+	}
+	return t
+}
+
+// IterRecords rewrites the IterRecords method of slowQueryTable.
+func (s slowQueryTable) IterRecords(ctx sessionctx.Context, startKey kv.Key, cols []*table.Column, fn table.RecordIterFunc) error {
+	dom := domain.GetDomain(ctx)
+	result := dom.ShowSlowQuery(&ast.ShowSlow{Tp: 42})
+
+	for i, item := range result {
+		row := make([]types.Datum, 0, len(cols))
+		row = append(row, types.NewDatum(item.SQL))
+
+		ts := types.NewTimeDatum(types.Time{types.FromGoTime(item.Start), mysql.TypeTimestamp, types.MaxFsp})
+		row = append(row, ts)
+		row = append(row, types.NewDurationDatum(types.Duration{item.Duration, types.MaxFsp}))
+		row = append(row, types.NewDatum(item.Detail.String()))
+		row = append(row, types.NewDatum(item.Succ))
+		row = append(row, types.NewDatum(item.ConnID))
+		row = append(row, types.NewDatum(item.TxnTS))
+		row = append(row, types.NewDatum(item.User))
+		row = append(row, types.NewDatum(item.DB))
+		row = append(row, types.NewDatum(item.TableIDs))
+		row = append(row, types.NewDatum(item.IndexIDs))
+		row = append(row, types.NewDatum(item.Internal))
```

Comment:
I found it's maybe useful to combine https://github.com/pingcap/parser/pull/32's sql digest into here, then we can agg a group similar sql's time.